python/ray/experimental/workflow/storage/__init__.py

```diff
@@ -11,21 +15,56 @@ def create_storage(storage_url: str) -> Storage:
 
     Args:
         storage_url: A URL indicates the storage type and root path.
+        Currently only two types of storages are supported: local fs and s3
+        For local fs, a path is needed, i.e.:
+           file:///local_path
+
+        For s3, bucket, path are necessary. In the meantime, other parameters
+        can be passed as well, like credientials or regions, i.e.:
+           s3://bucket/path?region_name=str&endpoint_url=str&aws_access_key_id=str&
+               aws_secret_access_key=str&aws_session_token=str
+
+        All parameters are optional and have the same meaning as boto3.client
 
     Returns:
         A storage instance.
     """
-    # TODO(suquark): in the future we need to support general URLs for
-    # different storages, e.g. "s3://xxxx/xxx". Currently we just use
-    # 'pathlib.Path' for convenience.
-    return FilesystemStorageImpl(storage_url)
+    parsed_url = parse.urlparse(storage_url)
+    if parsed_url.scheme == "file":
+        return FilesystemStorageImpl(parsed_url.path)
+    elif parsed_url.scheme == "s3":
+        bucket = parsed_url.netloc
+        s3_path = parsed_url.path
+        if not s3_path:
+            raise ValueError(f"Invalid s3 path: {s3_path}")
+        params = dict({
```

Comment:
We shouldn't encourage credentials to be passed via the URL. Instead, they should come from the environment.

Reply:
By default, it'll use the environment variables. 
In case the user wants to specify it manually, it can be passed through the URL. 
The URL here is just for an API to pass data around. It might not come from the os environment and the user can use 
`create_storage` and `set_global_storage` to set the storage.